applications/FluidDynamicsApplication/custom_processes/mass_conservation_check_process.cpp

```diff
@@ -606,6 +612,7 @@ void MassConservationCheckProcess::ShiftDistanceField( double deltaDist ){
     for(int count = 0; count < static_cast<int>(rNodes.size()); count++){
         ModelPart::NodesContainerType::iterator i_node = rNodes.begin() + count;
         i_node->FastGetSolutionStepValue( DISTANCE ) += deltaDist;
+        i_node->GetValue( DISTANCE_DIFFERENCE ) = -deltaDist;
```

Comment:
Considering that this correction is independent to what this process does, I wouldn't calculate the distance difference in here. Note that by doing so you oblige to use the mass conservation process to use the momentum correction.

Reply:
The application of the momentum correction is conditional with MOMENTUM_CORRECTION (being true or false). So, if one decides, he can deactivate it separately. On the other hand, having multiple conditions (for different processes) would make the code confusing (since at the moment the major change in the interface location comes from this process).

Reply:
This is not my point. My point is that (as far as I understand) this new term is independent on how you modify the level set. Imagine that I have a `custom_mass_conservation_check_process.h`. If this is the case your implementation will always add 0 correction.
My suggestion is to move the evaluation of the `DISTANCE_DIFFERENCE` outside of this process, namely to the `FluidAuxiliaryUtilities` and call it in the Python solver (if this is a problem, it can be temporary done in the solver in Python).
Having the calculation of this term hidden in a process, which might or might be not called (note that as it is right now it needs to be manually added to the json), is what makes the code confusing.

Reply:
Anyway, I do not have access to the difference outside the process. In order to fulfill your concern, I have added the condition to the mass conservation correction process as well. This makes tracking the instances of the momentum correction easier.

Reply:
OK, now I understand. For doing it as I suggest we would need to save the original distance in an auxiliary variable, namely the non-historical `DISTANCE` value.

Maybe we can rename `DISTANCE_DIFFERENCE` to `DISTANCE_CORRECTION` to make more clear the intended usage of this variable.

Reply:
Still, I do not think it is a good idea to do the comparison outside of the processes. Just consider that we have three different modifications (mass correction, re-distancing, etc.). it would be very hard to store the original distance somewhere between these processes if we want to do the momentum correction for example only for mass correction. In the current format, however, you can simply turn ON/OFF the MOMENTUM_CORRECTION wherever needed.

Reply:
Having said this, lets leave as it is right now.

Reply:
The variable is renamed to DISTANCE_CORRECTION.